src/common/SurgeStorage.cpp

```diff
@@ -1110,45 +1110,66 @@ void SurgeStorage::load_wt(int id, Wavetable *wt, OscillatorStorage *osc)
 {
     wt->current_id = id;
     wt->queue_id = -1;
+
     if (wt_list.empty() && id == 0)
     {
 #if HAS_JUCE
         load_wt_wt_mem(SurgeSharedBinary::memoryWavetable_wt,
                        SurgeSharedBinary::memoryWavetable_wtSize, wt);
 #endif
         if (osc)
-            strncpy(osc->wavetable_display_name, "Sin to Saw", TXT_SIZE);
+        {
+            osc->wavetable_display_name = "Sin to Saw";
+        }
 
         return;
     }
+
     if (id < 0)
+    {
         return;
+    }
+
     if (id >= wt_list.size())
+    {
         return;
+    }
+
     if (!wt)
+    {
         return;
+    }
 
     load_wt(path_to_string(wt_list[id].path), wt, osc);
 
     if (osc)
     {
-        auto n = wt_list.at(id).name;
-        strncpy(osc->wavetable_display_name, n.c_str(), 256);
+        osc->wavetable_display_name = wt_list.at(id).name;
     }
 }
 
 void SurgeStorage::load_wt(string filename, Wavetable *wt, OscillatorStorage *osc)
 {
```

Comment:
this is a fine change but we should open an issue to move this and a few other apis to fs::path